src/DeviceGraphicCard.php

```diff
@@ -140,83 +134,77 @@ public function prepareInputForAddOrUpdate($input)
         return $input;
     }
 
-
-    public function prepareInputForAdd($input)
+    public function prepareInputForAdd($input): array|false
     {
         return $this->prepareInputForAddOrUpdate($input);
     }
 
-
-    public function prepareInputForUpdate($input)
+    public function prepareInputForUpdate($input): array|false
```

Comment:
Very naive question: is that change will conflict with the signature of CommonDBTM::prepareInputForUpdate ?

Reply:
Methods in child classes can always have a more specific return type. Since the method in CommonDBTM has no return type specified (PHPDoc doesn't count), we can put anything here and not have conflicts. The only thing is if something extends this class, they need to either match this signature or make it even more specific (like removing `false` from the union type).